buildkite/scripts/rosetta-integration-tests.sh

```diff
@@ -2,103 +2,133 @@
+pg_ctlcluster ${POSTGRES_VERSION} main start
+pg_dropcluster --stop ${POSTGRES_VERSION} main
+pg_createcluster --start -d ${POSTGRES_DATA_DIR} --createclusterconf /rosetta/postgresql.conf ${POSTGRES_VERSION} main
+sudo -u postgres psql --command "CREATE USER ${POSTGRES_USERNAME} WITH SUPERUSER PASSWORD '${POSTGRES_USERNAME}';"
+sudo -u postgres createdb -O ${POSTGRES_USERNAME} ${POSTGRES_DBNAME}
+psql -f "${MINA_ARCHIVE_SQL_SCHEMA_PATH}" "${PG_CONN}"
 
 # Mina Rosetta
 echo "=========================== STARTING ROSETTA API ONLINE AND OFFLINE INSTANCES ==========================="
 ports=( $MINA_ROSETTA_ONLINE_PORT $MINA_ROSETTA_OFFLINE_PORT )
 for port in ${ports[*]}
 do
-    mina-rosetta${MINA_SUFFIX} \
+    mina-rosetta \
     --archive-uri "${PG_CONN}" \
     --graphql-uri http://127.0.0.1:${MINA_GRAPHQL_PORT}/graphql \
     --log-level ${LOG_LEVEL} \
     --port ${port} &
     sleep 5
 done
 
-# Configuration
-echo "=========================== DOWNLOADING CONFIGURATION FOR ${MINA_NETWORK} ==========================="
-curl -s -o "${MINA_CONFIG_FILE}" "https://raw.githubusercontent.com/MinaProtocol/mina/${MINA_COMMIT}/genesis_ledgers/${MINA_NETWORK}.json"
-
 # Mina Archive
 echo "========================= STARTING ARCHIVE NODE on PORT ${MINA_ARCHIVE_PORT} ==========================="
 mina-archive run \
-  --postgres-uri "${PG_CONN}" \
   --config-file ${MINA_CONFIG_FILE} \
   --log-level ${LOG_LEVEL} \
+  --postgres-uri "${PG_CONN}" \
   --server-port ${MINA_ARCHIVE_PORT} &
 sleep 5
 
-# Libp2p Keypair
-echo "=========================== GENERATING KEYPAIR IN ${MINA_LIBP2P_KEYPAIR_PATH} ==========================="
-mina libp2p generate-keypair -privkey-path ${MINA_LIBP2P_KEYPAIR_PATH}
-
 # Daemon
 echo "========================= STARTING DAEMON connected to ${MINA_NETWORK} ==========================="
-mina$MINA_SUFFIX daemon \
-  --libp2p-keypair ${MINA_LIBP2P_KEYPAIR_PATH} \
+mina daemon \
   --archive-address 127.0.0.1:${MINA_ARCHIVE_PORT} \
-  --peer-list-url ${PEER_LIST_URL} \
-  --external-port ${MINA_DAEMON_PORT} \
-  --rest-port ${MINA_GRAPHQL_PORT} \
-  --log-level ${LOG_LEVEL} \
+  --background \
+  --block-producer-pubkey "$BLOCK_PRODUCER_PK" \
   --config-file ${MINA_CONFIG_FILE} \
-  --background
-
-# wait for it to settle
-sleep 30
-
-# echo "========================= POPULATING MISSING BLOCKS ==========================="
-./download-missing-blocks.sh ${MINA_NETWORK} ${POSTGRES_DBNAME} ${POSTGRES_USERNAME} &
+  --libp2p-keypair ${MINA_LIBP2P_KEYPAIR_PATH} \
+  --log-level ${LOG_LEVEL} \
+  --proof-level none \
+  --rest-port ${MINA_GRAPHQL_PORT} \
+  --run-snark-worker "$SNARK_PRODUCER_PK" \
+  --seed
 
 echo "========================= WAITING FOR THE DAEMON TO SYNC ==========================="
 status="Pending"
 max_retries=10
 until [ $status == "Synced" ]
 do
   [[ $max_retries -eq 0 ]] && echo "Unable to Sync the Daemon" && exit 1  || ((max_retries--))
-  sleep 60
+  sleep 30
```

Comment:
I'd like something a little more reliable. Maybe:
```suggestion
while true; do
  if mina client status > /dev/null 2>&1; then
    break
  else
    sleep 1
  fi
done
```
Adding a limit of how long we're willing to wait might also be a good idea.

Reply:
the `if mina client status > /dev/null 2>&1; then` would only check if client can connect to the graphql interface.
From my tests, the status stays ~30 sec in `Bootstrap` before it is fully `Synced`. 

In the above loop, If the daemon does not start for some reasons, we are trapped in an infinite loop
I added a limit with the max_retries to offer a timeout / exit . The condition to move forward is the Daemon to return the status "Synced"

Reply:
Perhaps you can grep the output of `mina client status` to see if it's bootstrapped? If you think it doesn't make sense, feel free to ignore it, though.